src/database/migrations/20190819160153_create_tables.js

```diff
@@ -0,0 +1,63 @@
+exports.up = kn =>
+  kn.schema
+    .createTable("projects", tb => {
+      tb.increments();
+
+      tb.string("name")
+        .notNullable()
+        .unique();
+
+      tb.string("description");
+
+      tb.boolean("completed").defaultTo(false);
+    })
+    .createTable("resources", tb => {
+      tb.increments();
+
+      tb.string("name")
+        .notNullable()
+        .unique();
+
+      tb.string("description");
+    })
+    .createTable("project_resources", tb => {
+      tb.integer("project_id")
+        .unsigned()
+        .notNullable()
+        .references("projects.id")
+        .onDelete("CASCADE")
+        .onUpdate("CASCADE");
+
+      tb.integer("resource_id")
+        .unsigned()
+        .notNullable()
+        .references("resources.id")
+        .onDelete("CASCADE")
+        .onUpdate("CASCADE");
+
+      tb.primary(["project_id", "resource_id"]);
+    })
+    .createTable("tasks", tb => {
+      tb.increments();
+
+      tb.string("description").notNullable();
+
+      tb.string("notes");
+
+      tb.boolean("completed").defaultTo(false);
+
+      tb.integer("project_id")
+        .unsigned()
+        .notNullable()
+        .references("projects.id")
+        .onDelete("CASCADE")
+        .onUpdate("CASCADE");
+    });
+
+exports.down = kn => {
+  return kn.schema
+    .dropTableIfExists("tasks")
+    .dropTableIfExists("project_resources")
+    .dropTableIfExists("resources")
+    .dropTableIfExists("projects");
```

Comment:
the order of dropping the tables matters -- just like the order of providing the migrations tables matters. If we provide projects, resources, tasks and project_resources in that order, we have to drop in the reverse order: project_resources, tasks, resources, projects.